### PR TITLE
test: cover purchase event handler and migration edge cases

### PR DESCRIPTION
**Pull request by dai175**
## Summary
- Add unit tests for `PurchaseEventHandler` mixin (16 cases) — the revenue-critical path that updates subscription state after purchase completion/restoration was previously untested
- Add 2 integration scenarios for `DiaryService` encryption migration — document Hive CE silent data discard behavior when reopened with wrong key or without cipher (downgrade scenario)

## Test Plan
- [x] `fvm flutter test test/unit/services/purchase_event_handler_test.dart` — 16 passed
- [x] `fvm flutter test test/integration/diary_service_encryption_migration_test.dart` — 6 passed
- [x] `fvm flutter test` — 2016 total passed (existing 1998 + new 18)
- [x] `fvm flutter analyze` — No issues found
- [x] `fvm dart format .` — clean

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## テスト

* **Tests**
  * 暗号化マイグレーション関連の統合テストを追加しました。
  * 購入イベントハンドラーのユニットテストを追加しました。

[Review Change Stack](https://app.coderabbit.ai/change-stack/dai175/smart_photo_diary/pull/122?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: test/unit/services/purchase_event_handler_test.dart [hunk 1]
```diff
@@ -0,0 +1,508 @@
...
+      final purchase = _TestPurchaseDetails(
+        productID: 'unknown_invalid_product_id',
+        status: iap.PurchaseStatus.purchased,
+        pendingCompletePurchase: false,
+      );
+
+      await expectLater(
+        () => handler.applyPurchaseSideEffects(purchase),
+        throwsA(isA<Exception>()),
+      );
```
[the remark this answers is not in the document]
Comment (dai175) [reply test/unit/services/purchase_event_handler_test.dart:397]: 対応しました。`throwsA(isA<Exception>())` を `throwsA(isA<ServiceException>())` に変更しています（568b17c）。

File: test/unit/services/purchase_event_handler_test.dart [hunk 2]
```diff
@@ -0,0 +1,508 @@
...
+      final updatedStatus = captured.first as SubscriptionStatus;
+
+      expect(updatedStatus.planId, SubscriptionConstants.premiumYearlyPlanId);
+      final minExpiry = before.add(const Duration(days: 365));
+      expect(
+        updatedStatus.expiryDate!.isAfter(minExpiry) ||
+            updatedStatus.expiryDate!.isAtSameMomentAs(minExpiry),
+        isTrue,
+      );
+    });
```
[the remark this answers is not in the document]
Comment (dai175) [reply test/unit/services/purchase_event_handler_test.dart:310]: 対応しました。`after = DateTime.now()` と `maxExpiry` を追加し、上限チェック `isBefore(maxExpiry)` を追加しています（568b17c）。
